modules/jupyter-widget/src/widget-tooltip.js

```diff
@@ -0,0 +1,99 @@
+/* global document, console */
+let lastPickedObject;
+let lastTooltip;
+
+export default function getTooltip(pickedInfo) {
+  if (!pickedInfo.picked) {
+    return null;
+  }
+  if (pickedInfo.object === lastPickedObject) {
+    return lastTooltip;
+  }
+  const tooltip = {
+    html: tabularize(pickedInfo.object),
+    style: {
+      fontFamily: '"Helvetica Neue", Helvetica, Arial, sans-serif',
+      display: 'flex',
+      flex: 'wrap',
+      maxWidth: '500px',
+      flexDirection: 'column',
+      zIndex: 2
+    }
+  };
+  lastTooltip = tooltip;
+  lastPickedObject = pickedInfo.object;
+  return tooltip;
+}
+
+const EXCLUDES = new Set(['position', 'index']);
+
+function tabularize(json) {
+  // Turns a JSON object of picked info into HTML for a tooltip
+  const dataTable = document.createElement('div');
+  dataTable.className = 'dataTable';
+
+  // Creates rows of two columns for the tooltip
+  for (const key in json) {
+    if (EXCLUDES.has(key)) {
+      continue; // eslint-disable-line
+    }
+    const header = document.createElement('div');
+    header.className = 'header';
+    header.innerText = key;
+    const valueElement = document.createElement('div');
+    valueElement.className = 'value';
+
+    setInnerText(valueElement, json[key]);
```

Comment:
Nit: use the right level of abstraction:

`valueElement.innerText = toText(json[key])`